pyvex/block.py

```diff
@@ -114,52 +116,31 @@ def convert_expr(expr):
             """
             Converts a VEX expression to use tmps in the appended-block instead of the appended-to-block. Used to prevent
             collisions in tmp numbers between the two blocks.
-
             :param tmp:       The VEX expression to convert
             :vartype expr:    :class:`IRExpr`
             """
             if type(expr) is RdTmp:
                 expr.tmp = convert_tmp(expr.tmp)
 
-        #
-        # Mini handlers
-        #
-
-        def _handle_WrTmp(stmt): stmt.tmp = convert_tmp(stmt.tmp)
-        def _handle_LoadG(stmt): stmt.dst = convert_tmp(stmt.dst)
-        def _handle_LLSC(stmt): stmt.result = convert_tmp(stmt.result)
-        def _handle_Dirty(stmt):
-            if stmt.tmp not in invalid_vals:
-                stmt.tmp = convert_tmp(stmt.tmp)
-            for e in stmt.args:
-                convert_expr(e)
-        def _handle_CAS(stmt):
-            if stmt.oldLo not in invalid_vals: stmt.oldLo = convert_tmp(stmt.oldLo)
-            if stmt.oldHi not in invalid_vals: stmt.oldHi = convert_tmp(stmt.oldHi)
-
-        _mapping = {
-            LoadG: _handle_LoadG,
-            LLSC: _handle_LLSC,
-            Dirty: _handle_Dirty,
-            CAS: _handle_CAS,
-        }
-
         for stmt in extendwith.statements:
-
-            # special processing
-            if type(stmt) is WrTmp:
-                # This is the most frequent case
-                _handle_WrTmp(stmt)
-            else:
-                handler = _mapping.get(type(stmt), None)
-                if handler:
-                    handler(stmt)
-
-            # process all statements
+            stmttype = type(stmt)
+            if stmttype is WrTmp:
```

Comment:
Any good reason for us to switch back to cascading `if`s, instead of keep using the `dict`-based solution?

Reply:
The cascading `if`s are shorter, easier to read, and faster.